lisa/operating_system.py

```diff
@@ -1487,7 +1475,6 @@ class RPMDistro(Linux):
 
     # ex: dpdk-20.11-3.el8.x86_64 or dpdk-18.11.8-1.el7_8.x86_64
     _rpm_version_splitter_regex = re.compile(
-        r"(?P<package_name>[a-zA-Z0-9\-_]+)-"
```

Comment:
It may bring regressions.

Reply:
The new rpm query being used will no longer return package name in the result, but I agree that refactor can get a bit risky. Do you think it's possible to trigger test cases that are affected by this change to catch errors?

Reply:
This it needs to be compatible with different distro versions from AzureLinux 2.0 to other RPM based Linux. So it needs to keep for old versions.

Reply:
I will keep the regex patterns unmodified for this PR and do the refactor in a separate PR. Since the semver regex (*_version_splitter_regex) were the reason the parsing logic failed in the first place, so I still think they need to be updated. The dpkg and rpm queries I introduced should be consistent across distros so they are safe to use.